### ISSUE TITLE
Consider CIoU or Distance IoU instead of GIou for loss function?

### ISSUE THREAD

**Issue by lessw2020** [facebookresearch/detr#38]
In EfficientDet there was an improvement gain switching to Distance IoU, and suspect the same would hold for DETR with either Distance or Complete IoU.

>  By incorporating DIoU and CIoU losses into state-of-the-art object detection algorithms, e.g., YOLO v3, SSD and Faster RCNN, we achieve notable performance gains in terms of not only IoU metric but also GIoU metric. 

>  we consider three geometric factors, i.e., overlap area, normalized central point distance and aspect ratio, which are crucial for measuring bounding box regression in object detection and instance segmentation. 
The three geometric factors are then incorporated into CIoU loss for better distinguishing difficult regression cases. The training of deep models using CIoU loss results in consistent AP and AR improvements in comparison to widely adopted ℓn-norm loss and IoU-based loss.

Here's the paper discussing CIoU:
https://arxiv.org/abs/2005.03572
and Distance IoU:
https://arxiv.org/abs/1911.08287

and most importantly code:
https://github.com/Zzh-tju/CIoU

**Comment by fmassa** [comment]
Hi @lessw2020 ,

Thanks for the suggestion!
We use GitHub issues for [tracking bugs or unexpected behaviors in DETR codebase](https://github.com/facebookresearch/detr/issues/new/choose).
For research-related questions, please reach out by e-mail.

Thank you for your understanding!

**Comment by henbucuoshanghai** [comment]
then   using the diou or ciou?
